// load-cfg/dist/index.esm.js
import { readJsonSync } from 'fs-extra';
import { resolve, extname } from 'path';
import { sync } from 'find-up';
import { merge } from 'lodash/fp';

const loadFile = (filepath, noCache) => {
    require('@babel/register')({
        cache: !noCache,
        presets: [['@babel/preset-env', { modules: 'commonjs' }]],
    });
    let file;
    if (noCache && filepath) {
        delete require.cache[resolve(filepath)];
    }
    try {
        const isJS = extname(filepath) === '.js';
        if (isJS) {
            const required = require(filepath);
            file = required.default || required;
        }
        else {
            file = readJsonSync(filepath, { encoding: 'utf-8' });
        }
    }
    catch (err) {
        console.warn('There was an error loading your config:\n');
        console.warn(err);
    }
    return file;
};
const finds = (name) => [
    `${name}.json`,
    `.${name}rc`,
    `${name}rc.js`,
    `${name}rc.json`,
    `${name}.config.js`,
    `${name}.config.json`,
];
function load(name, defaultConfig, noCache, deep) {
    const filepath = sync(finds(name));
    const file = filepath ? loadFile(filepath, noCache) : {};
    const next = defaultConfig
        ? deep
            ? merge(defaultConfig, file)
            : Object.assign({}, defaultConfig, file)
        : file;
    // tslint:disable
    return next;
}
function loadFrom(filePath, defaultConfig, noCache, deep) {
    const file = loadFile(filePath, noCache);
    // tslint:disable
    return defaultConfig
        ? deep
            ? merge(defaultConfig, file)
            : Object.assign({}, defaultConfig, file)
        : file;
}

export { finds, load, loadFile, loadFrom };
